ContextualPopupDecorator/tests/ContextualPopupDecorator-specs.js

```diff
@@ -0,0 +1,168 @@
+		const handleClose = jest.fn();
+		const Root = FloatingLayerDecorator('div');
+		const message = 'goodbye';
+		render(
+			<Root>
+				<ContextualButton onClose={handleClose} open popupComponent={() => message}>
+					Hello
+				</ContextualButton>
+			</Root>
+		);
+		const contextualButton = screen.getByRole('button');
+
+		userEvent.click(contextualButton);
+
+		expect(handleClose).toHaveBeenCalled();
+	});
+
+	test('should render component into FloatingLayer if open', () => {
+		const Root = FloatingLayerDecorator('div');
+		const message = 'goodbye';
+		render(
+			<Root>
+				<ContextualButton open popupComponent={() => message}>
+					Hello
+				</ContextualButton>
+			</Root>
+		);
+		const contextualPopup = screen.getByRole('alert');
+
+		const expected = message;
+		const actual = contextualPopup.children.item(0);
+
+		expect(actual).toHaveTextContent(expected);
+	});
+
+	test('should not render into FloatingLayer if not open', () => {
+		const Root = FloatingLayerDecorator('div');
+		const message = 'goodbye';
+		render(
+			<Root>
+				<ContextualButton popupComponent={() => message}>
+					Hello
+				</ContextualButton>
+			</Root>
+		);
+
+		const popup = screen.queryByText(message);
+
+		expect(popup).toBeNull();
+	});
+
+	test('should not emit onClose event when clicking outside if noAutoDismiss is true', () => {
+		const handleClose = jest.fn();
+		const Root = FloatingLayerDecorator('div');
+		const message = 'goodbye';
+		render(
+			<Root data-testid="outsideArea">
+				<ContextualButton noAutoDismiss onClose={handleClose} open popupComponent={() => message}>
+					Hello
+				</ContextualButton>
+			</Root>
+		);
+		const outsideArea = screen.getByTestId('outsideArea');
+
+		userEvent.click(outsideArea);
+
+		expect(handleClose).not.toHaveBeenCalled();
+	});
+
+	test('should emit onClose event when clicking outside if noAutoDismiss is missing', () => {
+		const handleClose = jest.fn();
+		const Root = FloatingLayerDecorator('div');
+		const message = 'goodbye';
+		render(
+			<Root data-testid="outsideArea">
+				<ContextualButton onClose={handleClose} open popupComponent={() => message}>
+					Hello
+				</ContextualButton>
+			</Root>
+		);
+		const outsideArea = screen.getByTestId('outsideArea');
+
+		userEvent.click(outsideArea);
+
+		expect(handleClose).toHaveBeenCalled();
+	});
+
+	test('should have `right` className when direction is set to `right`', () => {
+		const handleClose = jest.fn();
+		const Root = FloatingLayerDecorator('div');
+		const message = 'goodbye';
+		render(
+			<Root>
+				<ContextualButton direction="right" onClose={handleClose} open popupComponent={() => message}>
+					Hello
+				</ContextualButton>
+			</Root>
+		);
+		const contextualPopup = screen.getByRole('alert').children.item(1);
+
+		const expectedAttribute = 'direction';
+		const expectedValue = 'right';
+
+		expect(contextualPopup).toHaveAttribute(expectedAttribute, expectedValue);
```

Comment:
the expectation is not the same as the description of the test. can you check the className ? Or update the description

Reply:
Description is wrong, will fix.